### ISSUE TITLE
BUG: negating pd.Na and `None`

### ISSUE THREAD

**Issue by attack68** [pandas-dev/pandas#42862]
- [x] I have checked that this issue has not already been reported.
- [ ] I have confirmed this bug exists on the latest version of pandas.
- [x] I have confirmed this bug exists on the master branch of pandas.

I don't really know how to describe this or if it is as expected but it seems strange behaviour to me:

```python
>>> df = pd.DataFrame({"a": [pd.NA, -1, None], "b": [np.nan, -1, 1]})
      a     b
0  <NA>   NaN
1    -1  -1.0
2  None   1.0

>>> print(df.dtypes)
a    object
b   float64

>>>print(df * -1)
     a    b
0  NaN  NaN
1    1  1.0
2  NaN -1.0

>>> print((df * -1).dtypes
a    object
b   float64

>>> print(-df)
TypeError: bad operand type for unary -: 'NoneType'
```
I guess there are two things here:
 1) `(-df)` is not the same as `(df * -1)`
 2) the operation keeps `dtype` as `object` but the `<NA>` and `None` are converted to `NaN` display

**Comment by KrishnaSai2020** [comment]
In df *-1 it defaults to NaN which is expected as None * -1 is invalid but surely it should do the same for -df.  but maybe because there's no * sign DataFrame.multiply doesn't register it.

**Comment by attack68** [comment]
> In df *-1 it defaults to NaN which is expected as None * -1 is invalid but surely it should do the same for -df. but maybe because there's no * sign DataFrame.multiply doesn't register it. So then Python kicks in and since in Python Multiplication of NaN and -1 is invalid it raises an error.

ok, but why is `pd.Na * -1 = np.nan`

**Comment by KrishnaSai2020** [comment]
for consistency maybe ie to make all values floats as pd.na stores an integer value

**Comment by KrishnaSai2020** [comment]
It seems strange but makes sense. Sort of anyway.

**Comment by attack68** [comment]
> So IMO object type data should match this and just propagate `NA`

I agree, I think `pd.NA = -pd.NA = pd.NA [*/+/-/**] float/int/complex`

**Comment by KrishnaSai2020** [comment]
+1 to redefining `pd.NA`

**Comment by mzeitlin11** [comment]
> +1 to redefining `pd.NA`

To be clear on the scope here, I'm not recommending a redefinition of `pd.NA`, this is how it already works. There are just problems that are specific to object type data
